### PR TITLE
Json2csv 

### PR DESCRIPTION
**Pull request by ashishpriyadarshiCIC**
Co-authored by @DumbMachine 
Further working on the work done by @DumbMachine  for GSoC 2020

### CODE REVIEW

**Comment by ashishpriyadarshiCIC** [comment]
@henrykironde I'm not squashing these commits for now, I'll do it just before it's merged

**Comment by henrykironde** [comment]
The style of code is missing something. try to run the code linters locally and see why they are failing. i.e  yapf and pylint

File: test/test_retriever.py [hunk 1]
```diff
@@ -517,21 +525,20 @@ def test_getmd5_path():
     exp_hash = '0bec5bf6f93c547bc9c6774acaf85e1a'
     assert getmd5(data=data_file, data_type='file') == exp_hash
 
-
-def test_json2csv():
+@pytest.mark.parametrize("test_name, json_data, header_values, row_key, expected", json2csv_datasets)
+def test_json2csv(test_name, json_data, header_values, row_key, expected):
     """Test json2csv function.
 
     Creates a json file and tests the md5 sum calculation.
```
Comment (apoorvaeternity) [inline test/test_retriever.py:532]: @ashishpriyadarshiCIC - Are we still calculating md5? Maybe we can update the comment if not.

**Comment by apoorvaeternity** [comment]
@ashishpriyadarshiCIC did you test this PR locally and is it ready for a review?

**Comment by henrykironde** [comment]
@ashishpriyadarshiCIC, if we could include commands that you run and some sample output

**Comment by henrykironde** [comment]
Ref: #1507

**Pull request closed without merging**